### DISCUSSION TITLE
[Simple question] Any available logger for test or validation steps? ＠Logger availability

### DISCUSSION THREAD

**Discussion by Hikaru-13915** [Lightning-AI/pytorch-lightning#10862] [code help: CV]
As all of you knows, we can use Loggers by setting like
`from pytorch_lightning.loggers import 'YOURLOGGER'`
`trainer = model.Trainer(logger=YOURLOGGER(save_dir="logs/", name="your_train"))`
Now I'm trying to use logger, CSVLogger, but it only tells me the information as to the training steps like bellow.
`  train_loss  train_acc  epoch  step  valid_loss  valid_acc  test_loss  test_acc`
`0    0.626578   0.875000      0    49         NaN        NaN        NaN       NaN`
`1    0.514625   0.875000      0    99         NaN        NaN        NaN       NaN`
`2    0.334852   0.906250      0   149         NaN        NaN        NaN       NaN`
`3    0.121677   0.953125      0   199         NaN        NaN        NaN       NaN`
`4    0.135196   0.968750      0   249         NaN        NaN        NaN       NaN`

The example above doesn't reveal "valid_loss" and "valid_acc",  "test_loss",  "test_acc", though I defined these values at their own steps in the model.

Is there any possible way to log a test or validation steps??
Please teach me any available Loggers for test steps and how to implement that

**Comment by Borda** [comment]
I think I can reproduce it in Kaggle notebook with `train_loss` so I see some number in progress bar but NaN in logger file - https://www.kaggle.com/code/jirkaborovec/cancer-subtype-baseline-with-lightning-torch
